src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -415,19 +415,19 @@ public void addHvdc(LfHvdc hvdc) {
         hvdcs.add(Objects.requireNonNull(hvdc));
     }
 
-    private static double dispatchQ(List<LfGenerator> generatorsThatControlVoltage, boolean reactiveLimits, double qToDispatch) {
+    protected static double dispatchQ(List<LfGenerator> generatorsThatControlVoltage, boolean reactiveLimits, double qToDispatch) {
         double residueQ = 0;
         double calculatedQ = qToDispatch / generatorsThatControlVoltage.size();
         Iterator<LfGenerator> itG = generatorsThatControlVoltage.iterator();
         while (itG.hasNext()) {
             LfGenerator generator = itG.next();
-            if (reactiveLimits && calculatedQ < generator.getMinQ()) {
-                generator.setCalculatedQ(generator.getCalculatedQ() + generator.getMinQ());
-                residueQ += calculatedQ - generator.getMinQ();
+            if (reactiveLimits && calculatedQ + generator.getCalculatedQ() < generator.getMinQ()) {
+                residueQ += calculatedQ + generator.getCalculatedQ() - generator.getMinQ();
```

Comment:
This residue Q must be correctly handled

Reply:
Yes but I dont know what is wrong ?

Reply:
residueQ must at the end be redistributed so that all the Q is correctly compensated, is that done when yes then all right

Reply:
well the thing is it sets the calculatedQ to the max or min bound so the residueQ needs to keep the CalculatedQ that was set on the generator before this. 
I can add a variable with the generator.getCalculatedQ() like that 
```java
if (reactiveLimits && calculatedQ + generator.getCalculatedQ() < generator.getMinQ()) {
                double generatorOldQ = generator.getCalculatedQ();
                generator.setCalculatedQ(generator.getMinQ());
                residueQ += calculatedQ + generatorOldQ - generator.getMinQ();
                
```

Reply:
I think this is problematic
What is the difference between calculatedQ and generator.getCalculatedQ()

Reply:
and even then you need to keep redistributing residueQ till its zero because it has to be conserved Q produced = QConsumed + Q losses

Reply:
calculatedQ is equal to dispatchQ/number of generators it is Q we ideally want to set to each generator at each loop.
generator.getCalculatedQ() is Q that was set previously to the generator in a previous loop because we try to dispatch Q with loops : 
```java
while (!generatorsThatControlVoltage.isEmpty() && Math.abs(qToDispatch) > Q_DISPATCH_EPSILON) {
            qToDispatch = dispatchQ(generatorsThatControlVoltage, reactiveLimits, qToDispatch);
}
```

Reply:
dispatchedQ is not really a good name it should be something like remainingQToDispatch

Reply:
I would suggest refactoring calculatedQ to tobedispatchedQ_pergen

Reply:
Here we need to include the factor for distributed slack  if any
so that the tobedispatchedQ_pergen would be multiplied by this but not in this PR i guess